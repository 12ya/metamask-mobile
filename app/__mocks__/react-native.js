import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
