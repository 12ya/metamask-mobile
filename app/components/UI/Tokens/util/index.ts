export { handleBalance } from './handleBalance';
export { sortAssets } from './sortAssets';
