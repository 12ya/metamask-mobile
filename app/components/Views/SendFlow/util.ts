import { toChecksumAddress } from 'ethereumjs-util';
import {useSelector} from 'react-redux';
import {
    selectNetwork,
    selectChainId
  } from '../../../selectors/networkController';

  import {
    validateAddressOrENS,
  } from '../../../util/address';

const identities = useSelector(
    (state: any) =>
      state.engine.backgroundState.PreferencesController.identities,
  );

  const addressBook = useSelector(
    (state: any) =>
    state.engine.backgroundState.AddressBookController.addressBook
    );

    const network = useSelector(selectNetwork);
    const chainId = useSelector(selectChainId);

const getAddressNameFromBookOrIdentities = (toAccount) => {
    if (!toAccount) return;

    const networkAddressBook = addressBook[network] || {};

    const checksummedAddress = toChecksumAddress(toAccount);

    return networkAddressBook[checksummedAddress]
      ? networkAddressBook[checksummedAddress].name
      : identities[checksummedAddress]
      ? identities[checksummedAddress].name
      : null;
  };

    /**
   * This set to the state all the information
   *  that come from validating an ENS or address
   * @param {*} toSelectedAddress - The address or the ens writted on the destination input
   */
  export const validateAddressOrENSFromInput = async ({toAccount}) => {
    console.log(toAccount, 'tosAccount');
    
    const {
      addressError,
      toEnsName,
      addressReady,
      toEnsAddress,
      addToAddressToAddressBook,
      toAddressName,
      errorContinue,
      isOnlyWarning,
      confusableCollection,
    } = await validateAddressOrENS({
      toAccount,
      network,
      addressBook,
      identities,
      chainId,
    });

    return null

    // return {addressError, toEnsName, toSelectedAddressReady: addressReady, toEnsAddressResolved: toEnsAddress, addToAddressToAddressBook, toSelectedAddressName: toAddressName, errorContinue, isOnlyWarning, confusableCollection}
  };

// export const onToSelectedAddressChange = ({toAccount, updateParentState}) => {

//     const addressName = getAddressNameFromBookOrIdentities(toAccount);
    
//     /**
//      * If the address is from addressBook or identities
//      * then validation is not necessary since it was already validated
//      */
//     if (addressName) {
//         updateParentState({toAccount, toSelectedAddressReady: true, isFromAddressBook: true, toSelectedAddressName: addressName})
//     } else {
//       validateAddressOrENSFromInput(toAccount);
//       /**
//        * Because validateAddressOrENSFromInput is an asynchronous function
//        * we are setting the state here synchronously, so it does not block the UI
//        * */
      
//       updateParentState({toAccount, isFromAddressBook: false, toSelectedAddressName: addressName, toSelectedAddressReady: false})
//     }
//   }; 