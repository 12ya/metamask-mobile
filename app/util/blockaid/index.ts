import { FlagType } from 'app/components/UI/BlockaidBanner/BlockaidBanner.types';

// eslint-disable-next-line import/prefer-default-export
export const showBlockaidUI = () =>
  process.env.MM_BLOCKAID_UI_ENABLED === 'true';

export const getAdditionalMetricsParams = (transaction: any) => {
  const { result_type, reason } = transaction.securityAlertResponse;
  let uiCustomizations;
  
  if (result_type === FlagType.Malicious) {
    uiCustomizations = ['flagged_as_malicious'];
  }

  const additionalParams: Record<string, any> = {
    ui_customizations: uiCustomizations,
  };

  if (result_type !== FlagType.Benign) {
    additionalParams['security_alert_response'] = result_type;
    additionalParams['security_alert_reason'] = reason;
  }

  return additionalParams;
};
