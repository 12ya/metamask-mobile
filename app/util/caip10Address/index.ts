import { isValidHexAddress } from '../address';

// Types for parsed CAIP-10 address components
export interface Caip10Address {
  accountId: string;
  chainId: string;
  address: string;
}

/**
 * Parses a CAIP-10 formatted address into its components.
 *
 * CAIP-10 addresses follow the format `namespace:reference:address`,
 * where `namespace` and `reference` identify the blockchain, and `address`
 * specifies the account on that blockchain. This function extracts and returns
 * these components as an object. It throws an error for invalid formats.
 *
 * @param {string} caip10Address - The CAIP-10 formatted address to parse.
 * @returns {Caip10Address} An object containing the parsed `accountId`, `chainId`,                          and `address`.
 * @throws {Error} If `caip10Address` does not conform to the CAIP-10 format.
 *
 * @example
 * parseCaip10Address('eip155:1:0x1234567890abcdefABCDEF1234567890ABCDEF');
 * // Returns: { accountId: 'eip155', chainId: '1', address: '0x1234567890abcdefABCDEF1234567890ABCDEF' }
 */
export function parseCaip10Address(caip10Address: string): Caip10Address {
  if (/\s/.test(caip10Address)) {
    throw new Error('CAIP-10 address format cannot contain spaces');
  }

  const parts = caip10Address.split(':');
  if (parts.length !== 3) {
    throw new Error('Invalid CAIP-10 address format');
  }
  const [accountId, chainId, address] = parts;

  // For now we only support EVM addresses, so we only validate those
  if (accountId === 'eip155') {
    if (!isValidHexAddress(address)) {
      throw new Error('Invalid Ethereum address format');
    }
  } else {
    throw new Error('Only support Ethereum addresses at the moment');
  }

  return { accountId, chainId, address };
}
