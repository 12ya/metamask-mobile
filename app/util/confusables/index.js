import { confusables } from 'unicode-confusables';
import { strings } from '../../../locales/i18n';
// import confusablesMap from 'unicode-confusables/data/confusables.json';

const confusablesMap = {
  A: 'Α',
  B: 'Β',
  C: 'Ϲ',
  D: 'Ꭰ',
  E: 'Ε',
  F: 'Ϝ',
  G: 'Ԍ',
  H: 'Η',
  I: 'Ι',
  J: 'Ј',
  K: 'Κ',
  L: 'Ꮮ',
  M: 'Μ',
  N: 'Ν',
  O: 'Ο',
  P: 'Ρ',
  Q: 'Ⴓ',
  R: 'Ꭱ',
  S: 'Ѕ',
  T: 'Τ',
  U: 'Ս',
  V: 'Ѵ',
  W: 'Ԝ',
  X: 'Χ',
  Y: 'Υ',
  Z: 'Ζ'
};

export const collectConfusables = (ensName) => {
  const key = 'similarTo';
  const collection = confusables(ensName).reduce(
    (total, current) => (key in current ? [...total, current.point] : total),
    [],
  );
  return collection;
};

const zeroWidthPoints = new Set([
  '\u200b', // zero width space
  '\u200c', // zero width non-joiner
  '\u200d', // zero width joiner
  '\ufeff', // zero width no-break space
  '\u2028', // line separator
  '\u2029', // paragraph separator,
]);

export const hasZeroWidthPoints = (char) => zeroWidthPoints.has(char);

export const getConfusablesExplanations = (confusableCollection) => [
  ...new Set(
    confusableCollection.map((key) => {
      const value = confusablesMap[key];
      return hasZeroWidthPoints(key)
        ? strings('transaction.contains_zero_width')
        : `'${key}' ${strings('transaction.similar_to')} '${value}'`;
    }),
  ),
];
