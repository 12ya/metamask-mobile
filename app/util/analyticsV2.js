import { InteractionManager } from 'react-native';
import DefaultPreference from 'react-native-default-preference';
import Analytics from '../core/Analytics/Analytics';
import Logger from './Logger';
import { DENIED, METRICS_OPT_IN } from '../constants/storage';

/**
 * This takes params with the following structure:
 * { foo : 'this is not anonymous', bar: {value: 'this is anonymous', anonymous: true} }
 * @param {Object} eventName
 * @param {Object} params
 */
export const trackEventV2 = (eventName, params) => {
  const init = async () => {
    const metricsOptIn = await DefaultPreference.get(METRICS_OPT_IN);
    if (metricsOptIn === DENIED) return;

    InteractionManager.runAfterInteractions(() => {
      let anonymousEvent = false;
      try {
        if (!params || Object.keys(params).length === 0) {
          Analytics.trackEvent(eventName);
        }

        const userParams = {};
        const anonymousParams = {};

        for (const key in params) {
          const property = params[key];

          if (
            property &&
            typeof property === 'object' &&
            !Array.isArray(property)
          ) {
            if (property.anonymous) {
              anonymousEvent = true;
              // Anonymous property - add only to anonymous params
              anonymousParams[key] = property.value;
            } else {
              // Non-anonymous property - add to both
              userParams[key] = property.value;
              anonymousParams[key] = property.value;
            }
          } else {
            // Non-anonymous properties - add to both
            userParams[key] = property;
            anonymousParams[key] = property;
          }
        }

        // Log all non-anonymous properties
        if (Object.keys(userParams).length) {
          Analytics.trackEventWithParameters(eventName, userParams);
        }

        // Log all anonymous properties
        if (anonymousEvent && Object.keys(anonymousParams).length) {
          Analytics.trackEventWithParameters(eventName, anonymousParams, true);
        }
      } catch (error) {
        Logger.error(error, 'Error logging analytics');
      }
    });
  };
  init();
};

export default {
  trackEvent: trackEventV2,
};
