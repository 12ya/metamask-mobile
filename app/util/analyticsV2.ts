import { InteractionManager } from 'react-native';
import { MetaMetrics, IMetaMetricsEvent } from '../core/Analytics';
import Logger from './Logger';

/**
 * This takes params with the following structure:
 * { foo : 'this is not anonymous', bar: {value: 'this is anonymous', anonymous: true} }
 * @param {Object} eventName
 * @param {Object} params
 */
const trackEventV2 = (event: IMetaMetricsEvent, params: any) => {
  InteractionManager.runAfterInteractions(() => {
    let anonymousEvent = false;
    try {
      if (!params && !event.properties) {
        MetaMetrics.trackEvent(event.name, {});
      }

      const userParams = { ...event.properties } as any;
      const anonymousParams = {} as any;

      for (const key in params) {
        const property = params[key];

        if (
          property &&
          typeof property === 'object' &&
          !Array.isArray(property)
        ) {
          if (property.anonymous) {
            anonymousEvent = true;
            // Anonymous property - add only to anonymous params
            anonymousParams[key] = property.value;
          } else {
            // Non-anonymous property - add to both
            userParams[key] = property.value;
            anonymousParams[key] = property.value;
          }
        } else {
          // Non-anonymous properties - add to both
          userParams[key] = property;
          anonymousParams[key] = property;
        }
      }

      // Log all non-anonymous properties
      if (Object.keys(userParams).length) {
        MetaMetrics.trackEvent(event.name, userParams);
      }

      // Log all anonymous properties
      if (anonymousEvent && Object.keys(anonymousParams).length) {
        MetaMetrics.trackAnonymousEvent(event.name, anonymousParams);
      }
    } catch (error: any) {
      Logger.error(error, 'Error logging analytics');
    }
  });
};

/**
 * This functions logs errors to analytics instead of sentry.
 * The objective is to log errors (that are not errors from our side) like “Invalid Password”.
 * An error like this generally means a user inserted the wrong password, so logging to sentry doesn't make sense.
 * But we still want to log this to analytics so that we are aware of a rapid increase which may mean it's an error from our side, for example, an error with the encryption library.
 * @param {String} type
 * @param {String} errorMessage
 * @param {String} otherInfo
 */
const trackErrorAsAnalytics = (
  type: string,
  errorMessage: string,
  otherInfo: string,
) => {
  try {
    MetaMetrics.trackEvent('Error occurred', {
      error: true,
      type,
      errorMessage,
      otherInfo,
    });
  } catch (error: any) {
    Logger.error(error, 'Error logging analytics - trackErrorAsAnalytics');
  }
};

export default {
  trackEvent: trackEventV2,
  trackErrorAsAnalytics,
};
