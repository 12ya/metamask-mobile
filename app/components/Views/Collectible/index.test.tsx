import React from 'react';
import { render } from '@testing-library/react-native';
import Collectible from './';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { backgroundState } from '../../../util/test/initial-root-state';
import { ThemeContext } from '../../../util/theme';

// Mock navigation object
const mockNavigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  // Add other navigation functions that might be used by the Collectible component or its children
};

// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => mockNavigation,
}));

// Updated mock ThemeContext to provide a comprehensive 'colors' properties for the Collectible component
const theme = {
  colors: {
    primary: {
      default: '#000', // Example color value, replace with actual theme primary color
    },
    background: {
      default: '#FFF', // Example color value, replace with actual theme background color
    },
    icon: {
      default: '#000', // Example color value, replace with actual theme icon color
    },
    overlay: {
      default: '#000', // Example color value, replace with actual theme overlay color
    },
    border: {
      muted: '#000', // Example color value, replace with actual theme border muted color
    },
    text: {
      default: '#000', // Example color value, replace with actual theme text default color
    },
    error: {
      default: '#FF0000', // Added error color
    },
    warning: {
      default: '#FFA500', // Added warning color
    },
    success: {
      default: '#008000', // Added success color
    },
    info: {
      default: '#0000FF', // Added info color
    },
    // Add any other color properties that might be used by the component
  },
};

const mockStore = configureMockStore();
const initialState = {
  engine: {
    backgroundState,
  },
  modals: {
    collectibleContractModalVisible: false,
  },
};
const store = mockStore(initialState);

describe('Collectible', () => {
  it('should render correctly', () => {
    const { toJSON } = render(
      <ThemeContext.Provider value={theme}>
        <Provider store={store}>
          <Collectible
            route={{ params: { address: '0x1' } }}
            navigation={mockNavigation}
          />
        </Provider>
      </ThemeContext.Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
