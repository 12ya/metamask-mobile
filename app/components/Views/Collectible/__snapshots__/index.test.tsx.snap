// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collectible should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        colors={
          [
            "#000",
          ]
        }
        onRefresh={[Function]}
        refreshing={false}
        tintColor="#000"
      />
    }
    style={
      {
        "backgroundColor": "#FFF",
        "flex": 1,
      }
    }
  >
    <RCTRefreshControl />
    <View>
      <View>
        <View>
          <View
            style={
              {
                "alignContent": "center",
                "alignItems": "center",
                "borderBottomColor": "#000",
                "borderBottomWidth": 0.5,
                "flex": 1,
                "paddingBottom": 30,
                "paddingHorizontal": 20,
              }
            }
            testID="collectible-overview-screen"
          >
            <View
              style={
                {
                  "marginTop": 20,
                }
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "#undefined",
                    "borderRadius": 12,
                    "flex": 0,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": undefined,
                        "borderRadius": 8,
                        "justifyContent": "flex-start",
                      },
                      undefined,
                      undefined,
                      {
                        "height": 50,
                        "width": 50,
                      },
                      undefined,
                      undefined,
                    ]
                  }
                  testID="fallback-nft-with-token-id"
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#000",
                          "fontFamily": "EuclidCircularB-Regular",
                          "fontSize": 30,
                          "fontWeight": "400",
                          "marginVertical": 2,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "fontSize": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "marginTop": 16,
                          "textAlign": "center",
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 20,
                  "marginTop": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#000",
                    "fontFamily": "EuclidCircularB-Regular",
                    "fontSize": 30,
                    "fontWeight": "400",
                    "textAlign": "center",
                  }
                }
                testID="collectible-name"
              >
                0
                 
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "width": "50%",
                }
              }
            >
              <TouchableOpacity
                onPress={[Function]}
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexShrink": 1,
                      "justifyContent": "center",
                      "marginHorizontal": 0,
                      "minWidth": 60,
                    },
                    undefined,
                  ]
                }
                testID="token-send-button"
              >
                <View
                  style={
                    {
                      "alignContent": "center",
                      "alignItems": "center",
                      "backgroundColor": "#0376c9",
                      "borderRadius": 18,
                      "height": 36,
                      "justifyContent": "center",
                      "paddingLeft": 1,
                      "paddingTop": 4,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 20,
                        },
                        {
                          "alignContent": "center",
                          "color": "#ffffff",
                          "justifyContent": "center",
                          "textAlign": "center",
                        },
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#000",
                        "fontFamily": "EuclidCircularB-Regular",
                        "fontSize": 30,
                        "fontWeight": "400",
                        "marginVertical": 2,
                      },
                      {
                        "textAlign": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#0376c9",
                        "fontSize": 14,
                        "marginHorizontal": 3,
                        "marginTop": 8,
                      },
                    ]
                  }
                >
                  Send
                </Text>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={[Function]}
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexShrink": 1,
                      "justifyContent": "center",
                      "marginHorizontal": 0,
                      "minWidth": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignContent": "center",
                      "alignItems": "center",
                      "backgroundColor": "#0376c9",
                      "borderRadius": 18,
                      "height": 36,
                      "justifyContent": "center",
                      "paddingLeft": 1,
                      "paddingTop": 4,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 30,
                        },
                        {
                          "alignContent": "center",
                          "color": "#ffffff",
                          "justifyContent": "center",
                          "textAlign": "center",
                        },
                        {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#000",
                        "fontFamily": "EuclidCircularB-Regular",
                        "fontSize": 30,
                        "fontWeight": "400",
                        "marginVertical": 2,
                      },
                      {
                        "textAlign": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#0376c9",
                        "fontSize": 14,
                        "marginHorizontal": 3,
                        "marginTop": 8,
                      },
                    ]
                  }
                >
                  Add
                </Text>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={[Function]}
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexShrink": 1,
                      "justifyContent": "center",
                      "marginHorizontal": 0,
                      "minWidth": 60,
                    },
                    undefined,
                  ]
                }
                testID="collectible-info-button"
              >
                <View
                  style={
                    {
                      "alignContent": "center",
                      "alignItems": "center",
                      "backgroundColor": "#0376c9",
                      "borderRadius": 18,
                      "height": 36,
                      "justifyContent": "center",
                      "paddingLeft": 1,
                      "paddingTop": 4,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 30,
                        },
                        {
                          "alignContent": "center",
                          "color": "#ffffff",
                          "justifyContent": "center",
                          "textAlign": "center",
                        },
                        {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#000",
                        "fontFamily": "EuclidCircularB-Regular",
                        "fontSize": 30,
                        "fontWeight": "400",
                        "marginVertical": 2,
                      },
                      {
                        "textAlign": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#0376c9",
                        "fontSize": 14,
                        "marginHorizontal": 3,
                        "marginTop": 8,
                      },
                    ]
                  }
                >
                  Info
                </Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFF",
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#FFF",
                "flex": 1,
              }
            }
            testID="collectibles"
          >
            <RCTScrollView
              refreshControl={
                <RefreshControlMock
                  colors={
                    [
                      "#000",
                    ]
                  }
                  refreshing={false}
                  tintColor="#000"
                />
              }
            >
              <RCTRefreshControl />
              <View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#FFF",
                      "justifyContent": "center",
                      "marginTop": 50,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": undefined,
                        "fontFamily": "EuclidCircularB-Regular",
                        "fontSize": 20,
                        "fontWeight": "400",
                      }
                    }
                  >
                    Don’t see your NFT?
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    onSwipeComplete={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />
</View>
`;
